Tools/CMake/AMReX_Config.cmake

```diff
@@ -141,11 +141,22 @@ function (configure_amrex)
 
    if ( BUILD_SHARED_LIBS OR AMReX_CUDA )
       if(APPLE)
-         target_link_options(amrex PUBLIC -Wl,-undefined,warning)
+        target_link_options(amrex PUBLIC "LINKER:-undefined,warning")
       elseif(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC" OR
              CMAKE_CXX_SIMULATE_ID STREQUAL "MSVC")
+
       else()
-         target_link_options(amrex PUBLIC -Wl,--warn-unresolved-symbols)
+
+        set(host_link_supported OLD)
+        if(CMP0105)
+          cmake_policy(GET CMP0105 host_link_supported)
+        endif()
+
+        if( host_link_supported STREQUAL "NEW" )
+          target_link_options(amrex PUBLIC "$<HOST_LINK:LINKER:--warn-unresolved-symbols>")
+        else()
+          target_link_options(amrex PUBLIC "LINKER:--warn-unresolved-symbols")
```

Comment:
Maybe rolling this line back to
```
target_link_options(amrex PUBLIC "-Wl,--warn-unresolved-symbols")
```
could solve the regression? @robertmaynard

Reply:
That will break CMake 3.18+ as the `--warn-unresolved-symbols` will be propagated to the device link step for nvcc.

Reply:
@mic84 is it possible that we patch-out the generator-expression?

Reply:
> That will break CMake 3.18+ ...

Not sure, since this line (158) is not taken for CMake 3.18+ (it uses 156 instead, no?)

Reply:
Sorry, I misread your first comment @ax3l. I thought you meant roll back both the `if` and `else`. I think rolling back `LINKER:` would be sufficient.